src/AzureAuth/Ado/AdoToken.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth.Ado
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Microsoft.Office.Lasso.Interfaces;
+
+    /// <summary>
+    /// Business logic for either getting an ADO PAT from <see cref="IEnv"/>

```

Comment:
[nit] Since our code lives in github, we might want to update ADO usages to AzureDevOps?

Reply:
`Azure DevOps` (with a space between words) would be better, but I also think ADO is a common enough abbreviation for us to get away with it here, especially for an internal comment rather than user-facing text. I could go either way. 🤷‍♂️

Reply:
Yeah if we think ADO is known, then it's cool.

Reply:
The public command is going to be `ado` with the name in the help, so I don't think we can make the argument that the full name is needed here, and not there. One could argu the command should be `azure-devops`, but I think the commands being short, easy to type, and are well enough understood we don't need that, + we do spell it out in the help text.